tools/rgbasm_version.asm

```diff
@@ -0,0 +1,65 @@
+IF (DEF(__RGBDS_MAJOR__) && DEF(__RGBDS_MINOR__) && DEF(__RGBDS_PATCH__))
+    ; RGBDS 0.3.3 introduced the '__RGBDS_MAJOR/MINOR/PATCH__' version constants
+    IF (__RGBDS_MAJOR__ > 0 || __RGBDS_MINOR__ >= 4)
+        ; RGBDS 0.4.0 introduced the `d:` format specifier
+        IF (__RGBDS_MAJOR__ > 0 || __RGBDS_MINOR__ >= 5)
+            ; RGBDS 0.5.0 introduced the 'PRINT' directive
+            IF (DEF(MAJOR))
+                PRINT "{d:__RGBDS_MAJOR__}"
+            ENDC
+            IF (DEF(MINOR))
+                PRINT "{d:__RGBDS_MINOR__}"
+            ENDC
+            IF (DEF(PATCH))
+                PRINT "{d:__RGBDS_PATCH__}"
+            ENDC
+        ELSE
+            ; RGBDS <0.5.0 used the 'PRINTT' directive
+            IF (DEF(MAJOR))
+                PRINTT "{d:__RGBDS_MAJOR__}"
+            ENDC
+            IF (DEF(MINOR))
+                PRINTT "{d:__RGBDS_MINOR__}"
+            ENDC
+            IF (DEF(PATCH))
+                PRINTT "{d:__RGBDS_PATCH__}"
+            ENDC
+        ENDC
+    ELSE
+        ; RGBDS <0.4.0 could only print hex numbers
+        IF (DEF(MAJOR))
+            PRINTT STRSUB("012345678910", __RGBDS_MAJOR__ + 1, 1 + (__RGBDS_MAJOR__ > 9))
+        ENDC
+        IF (DEF(MINOR))
+            PRINTT STRSUB("012345678910", __RGBDS_MINOR__ + 1, 1 + (__RGBDS_MINOR__ > 9))
+        ENDC
+        IF (DEF(PATCH))
+            PRINTT STRSUB("012345678910", __RGBDS_PATCH__ + 1, 1 + (__RGBDS_PATCH__ > 9))
+        ENDC
+    ENDC
+ELSE
+    ; RGBASM 0.3.9 was missing the '__RGBDS_MAJOR/MINOR/PATCH__' version constants
```

Comment:
I may have gone a bit too far with making this script as compatible as possible. :3

Reply:
This would be sufficient, since you don't care about distinctions below 0.5.0:
```asm
IF (DEF(__RGBDS_MAJOR__) && DEF(__RGBDS_MINOR__) && DEF(__RGBDS_PATCH__))
    IF (__RGBDS_MAJOR__ > 0 || __RGBDS_MINOR__ >= 5)
        IF (DEF(MAJOR))
            PRINT "{d:__RGBDS_MAJOR__}"
        ENDC
        IF (DEF(MINOR))
            PRINT "{d:__RGBDS_MINOR__}"
        ENDC
        IF (DEF(PATCH))
            PRINT "{d:__RGBDS_PATCH__}"
        ENDC
    ELSE
        PRINTT "0"
    ENDC
ELSE
    PRINTT "0"
ENDC
```

Reply:
yup, I think the smaller version should be better.

Although, once #555 is merged, we could do the minimum rgbds detection in an early asm file (rather than in the makefile). This would make this file entirely unecessary.

Should we way for #555 to be merged, and then try to remove this file completely?

Reply:
Semi-related question: how do you feel about indenting the contents of assembly blocks? RGBDS 0.5.0 introduced `DEF` syntax for defining constants, which improves on the old `x EQU y` syntax by being indentable (and in 0.7.0 the old syntax is deprecated). I want to PR the `DEF` syntax anyway, and would like to know if I should change things like this:
```asm
macro SET_DEFAULT
IF !DEF(\1)
\1 equ \2
ENDC
endm
```
to this:
```asm
macro SET_DEFAULT
    IF !DEF(\1)
        def \1 equ \2
    ENDC
endm
```